presto-bytecode/src/main/java/com/facebook/presto/bytecode/CompilerUtils.java

```diff
@@ -48,18 +53,26 @@
     private static final boolean RUN_ASM_VERIFIER = false; // verifier doesn't work right now
     private static final AtomicReference<String> DUMP_CLASS_FILES_TO = new AtomicReference<>();
     private static final AtomicLong CLASS_ID = new AtomicLong();
+    private static final DateTimeFormatter TIMESTAMP_FORMAT = DateTimeFormatter.ofPattern("YYYYMMdd_HHmmss");
 
     private CompilerUtils()
     {
     }
 
-    public static ParameterizedType makeClassName(String baseName)
+    public static ParameterizedType makeClassName(String baseName, Optional<String> suffix)
     {
-        String className = baseName + "_" + CLASS_ID.incrementAndGet();
+        String className = baseName + "_"
+                + suffix.orElse(Instant.now().atZone(UTC).format(TIMESTAMP_FORMAT))
```

Comment:
Note this will eagerly materialize the parameter `Instant.now().atZone(UTC).format(TIMESTAMP_FORMAT)` even `suffix` is presented.

I don't think it's a major performance issue since `suffix` will only be presented in two places. @haozhun  any thoughts how should we handle this?